lib/dialects/mssql/query-generator.js

```diff
@@ -333,18 +333,32 @@ const QueryGenerator = {
 
 
     //Obtain primaryKeys, uniquekeys and identity attrs from rawAttributes as model is not passed
-    for (const key in rawAttributes) {
-      if (rawAttributes[key].primaryKey) {
-        primaryKeysAttrs.push(rawAttributes[key].field || key);
+    for (const key in model.rawAttributes) {
+      if (model.rawAttributes[key].primaryKey) {
+        primaryKeysAttrs.push(model.rawAttributes[key].field || key);
       }
-      if (rawAttributes[key].unique) {
-        uniqueAttrs.push(rawAttributes[key].field || key);
+      if (model.rawAttributes[key].unique) {
+        uniqueAttrs.push(model.rawAttributes[key].field || key);
       }
-      if (rawAttributes[key].autoIncrement) {
-        identityAttrs.push(rawAttributes[key].field || key);
+      if (model.rawAttributes[key].autoIncrement) {
+        identityAttrs.push(model.rawAttributes[key].field || key);
       }
     }
 
+    model.options.indexes.forEach(index => {
+      //Add unique indexes defined by indexes option to uniqueAttrs
+      if (index.unique && index.fields) {
+        for (const field of index.fields) {
+          const fieldName = typeof field === 'string' ? field : field.name || field.attribute;
+          if (uniqueAttrs.indexOf(fieldName) === -1 && model.rawAttributes[fieldName]) {
+            uniqueAttrs.push(fieldName);
+          }
+        }
+      }
+
+      return index;
+    });
+
```

Comment:
@alitaheri Since you are not modifying the `model.options.indexes` array, its better to use `for..of`. Also, no need to return the index.